class Solution {
    public int characterReplacement(String s, int k) {
        int n=s.length();
        int l=0;
        int r=0;
        int maxlen=0;
        int maxf=0;
        int hash[]=new int[26];
        while(r<n){
hash[s.charAt(r)-'A']++;
maxf=Math.max(maxf,hash[s.charAt(r)-'A']);
while((r-l+1)-maxf>k){
hash[s.charAt(l)-'A']--;
maxf=0;
for(int j=0;j<25;j++){
maxf=Math.max(maxf,hash[j]);
}
l++;
}
if((r-l+1)-maxf<=k) maxlen=Math.max(maxlen,(r-l+1));
r++;
        }
        return maxlen;
    }
}